framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -205,6 +225,32 @@ public void updateForMethodCall(
             MethodInvocationNode n, AnnotatedTypeFactory atypeFactory, V val) {
         ExecutableElement method = n.getTarget().getMethod();
 
+        // List of expressions that this method side-effects (specified as arguments/elements of
+        // @SideEffectsOnly). If the list is empty, then there is no @SideEffectsOnly annotation.
+        List<JavaExpression> sideEffectsOnlyExpressions = new ArrayList<>();
+        if (isSideEffectsOnly(atypeFactory, method)) {
+            SourceChecker checker = analysis.checker;
+            JavaExpressionParseUtil.JavaExpressionContext methodUseContext =
+                    JavaExpressionParseUtil.JavaExpressionContext.buildContextForMethodUse(
+                            n, checker);
+
+            AnnotationMirror sefOnlyAnnotation =
```

Comment:
The call to method `isSideEffectsOnly` computes this, and now this computes it again.  Why not compute this before the `if` and have the `if` test whether the variable is null?  Then you don't need the `isSideEffectsOnly` method.